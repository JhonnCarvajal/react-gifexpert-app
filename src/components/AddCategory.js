import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");
  const handleInputChange = (e) => {
    //console.log(e.target.value);
    setInputValue(e.target.value);
    //console.log("Handle input Change llamado");
  };
  const hanldeSubmit = (e) => {
    e.preventDefault();
    //console.log("Handle submit",inputValue);
    if (inputValue.trim().length > 1) {
      setCategories((cats) => [inputValue, ...cats]); //se paso como props una funcion(setCategories) aqui se puede ver la utilidad
      // de usar el callback del use State para acceder al estado anterior, pues no contamos con ese estado directamente, al ser otro componente
      setInputValue("");
    }
  };
  return (
    <form onSubmit={hanldeSubmit}>
      <p>{inputValue}</p>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};
AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
