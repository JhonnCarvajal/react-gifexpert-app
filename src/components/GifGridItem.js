import React from "react";

export const GifGridItem = (props) => {
  //console.log(props);
  return <div className={'card animate__animated animate__fadeIn'}>
      {props.title}
      <img src={props.url} alt = 'Cargando...'/>
      </div>;
};
