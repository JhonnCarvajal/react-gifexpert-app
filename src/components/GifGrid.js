import React from "react";
import { useFetcGifs } from "../hook/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";
export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetcGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loading && <p>Loading...</p>}
      <div className="card-grid animate__animated animate__fadeIn">
        {images.map((img) => {
          return <GifGridItem key={img.id} {...img} />;
        })}
      </div>
    </>
  );
};
GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
