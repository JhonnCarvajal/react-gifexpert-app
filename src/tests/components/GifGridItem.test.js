import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Probar el GifGridItem", () => {
  const title = "Mi titulo";
  const url =
    "https://elcomercio.pe/resizer/3rNqVB9iUBkT2Y1LIS2i6qSyuro=/580x330/smart/filters:format(jpeg):quality(75)/arc-anglerfish-arc2-prod-elcomercio.s3.amazonaws.com/public/UQO2N3UJQFCVTASI64CXYMKS6U.jpg";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);
  test("Debe rendereizar correctamente el componente", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("Debe de tener un parrafo con el title", () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });
  test("La imagen debe tener src y alt", () => {
    const img = wrapper.find('img');
    //console.log(img.props());
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });
  test('Debe tener css, animated fadeIn', () => {
    const div = wrapper.find('div');
    //console.log(div.props());
    expect(div.prop('className').includes('animate__fadeIn')).toBe(true);
      
  })
  
});
