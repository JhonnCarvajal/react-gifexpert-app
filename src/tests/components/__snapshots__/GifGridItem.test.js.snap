// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar el GifGridItem Debe rendereizar correctamente el componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <p>
    Mi titulo
  </p>
  <img
    alt="Mi titulo"
    src="https://elcomercio.pe/resizer/3rNqVB9iUBkT2Y1LIS2i6qSyuro=/580x330/smart/filters:format(jpeg):quality(75)/arc-anglerfish-arc2-prod-elcomercio.s3.amazonaws.com/public/UQO2N3UJQFCVTASI64CXYMKS6U.jpg"
  />
</div>
`;
